src/modules/dimensionTypes.js

```diff
@@ -0,0 +1,56 @@
+/**ARTIFICIAL**/
+export const DIMENSION_TYPE_ALL = 'ALL'
+
+/**PROGRAM**/
+export const DIMENSION_TYPE_DATA_ELEMENT = 'DATA_ELEMENT'
+export const DIMENSION_TYPE_PROGRAM_ATTRIBUTE = 'PROGRAM_ATTRIBUTE'
+export const DIMENSION_TYPE_PROGRAM_INDICATOR = 'PROGRAM_INDICATOR'
+export const DIMENSION_TYPE_CATEGORY = 'CATEGORY'
+export const DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET =
+    'CATEGORY_OPTION_GROUP_SET'
+
+export const DIMENSION_TYPES_PROGRAM = new Set([
+    DIMENSION_TYPE_DATA_ELEMENT,
+    DIMENSION_TYPE_PROGRAM_ATTRIBUTE,
+    DIMENSION_TYPE_PROGRAM_INDICATOR,
+    DIMENSION_TYPE_CATEGORY,
+    DIMENSION_TYPE_CATEGORY_OPTION_GROUP_SET,
+])
+
+/**YOURS**/
+export const DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET =
+    'ORGANISATION_UNIT_GROUP_SET'
+
+export const DIMENSION_TYPES_YOURS = new Set([
+    DIMENSION_TYPE_ORGANISATION_UNIT_GROUP_SET,
+])
+
+/**MAIN**/
+export const DIMENSION_TYPE_OU = 'ou'
+export const DIMENSION_TYPE_PROGRAM_STATUS = 'programStatus'
+export const DIMENSION_TYPE_EVENT_STATUS = 'eventStatus'
+export const DIMENSION_TYPE_STORED_BY = 'storedBy'
+export const DIMENSION_TYPE_LAST_UPDATED_BY = 'lastUpdatedBy'
+
+export const DIMENSION_TYPES_MAIN = new Set([
+    DIMENSION_TYPE_OU,
+    DIMENSION_TYPE_PROGRAM_STATUS,
+    DIMENSION_TYPE_EVENT_STATUS,
+    DIMENSION_TYPE_STORED_BY,
+    DIMENSION_TYPE_LAST_UPDATED_BY,
+])
+
+/**TIME**/
+export const DIMENSION_TYPE_EVENT_DATE = 'eventDate'
+export const DIMENSION_TYPE_ENROLLMENT_DATE = 'enrollmentDate'
+export const DIMENSION_TYPE_INCIDENT_DATE = 'incidentDate'
+export const DIMENSION_TYPE_SCHEDULED_DATE = 'scheduledDate'
+export const DIMENSION_TYPE_LAST_UPDATED = 'lastUpdated'
+
+export const DIMENSION_TYPES_TIME = new Set([
```

Comment:
These are also defined in `modules/timeDimensions`. We should consolidate somehow.